promql/ast.go

```diff
@@ -319,3 +337,82 @@ func Children(node Node) []Node {
 		panic(errors.Errorf("promql.Children: unhandled node type %T", node))
 	}
 }
+
+// PositionRange describes a position in the input string of the parser.
+type PositionRange struct {
+	Start Pos
+	End   Pos
+}
+
+// mergeRanges is a helper function the PostionRanges of two Nodes
+// Note that the arguments must already be in the same order as they
```

Comment:
must be in the order they occur in the input string


The "already" implies you're passing a slice.

Reply:
Thanks